pkg/client/client_v2.go

```diff
@@ -118,11 +139,13 @@ type clientV2 struct {
 	cfgMu  sync.RWMutex
 	obsMu  sync.RWMutex
 
-	kickCh  chan struct{}
-	errCh   chan error
-	unitsCh chan UnitChanged
-	unitsMu sync.RWMutex
-	units   []*Unit
+	kickSendObservedCh chan struct{}
+	errCh              chan error
+	changesCh          chan UnitChanged
+	featuresMu         sync.Mutex
+	features           Features
```

Comment:
Going to need to add the `featuresIdx` here as well. I would start the initial index at 1.

There is a reason for starting it at 1, and that is backwards compatibility. With older clients, I would modify the Elastic Agent to read the observed features idx at 0, as the client doesn't understand features and we should not trigger an `unsettle` when it doesn't match, because it will never match. If it gets a 1, then it knows its a client that supports it but doesn't know about it. Or you add a boolean field in the observed message to signal that the client supports features.

Reply:
Unless I'm missing something, I'm not sure we need `features` (and related fields) here on the `clientV2` struct at all, since `features` is a field on the `Unit` struct (similar to `logLevel`).